aws/resource_aws_glue_trigger.go

```diff
@@ -0,0 +1,457 @@
+	"log"
+	"time"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/glue"
+	"github.com/hashicorp/terraform/helper/resource"
+	"github.com/hashicorp/terraform/helper/schema"
+	"github.com/hashicorp/terraform/helper/validation"
+)
+
+func resourceAwsGlueTrigger() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceAwsGlueTriggerCreate,
+		Read:   resourceAwsGlueTriggerRead,
+		Update: resourceAwsGlueTriggerUpdate,
+		Delete: resourceAwsGlueTriggerDelete,
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+		Timeouts: &schema.ResourceTimeout{
+			Create: schema.DefaultTimeout(5 * time.Minute),
+			Delete: schema.DefaultTimeout(5 * time.Minute),
+		},
+
+		Schema: map[string]*schema.Schema{
+			"actions": {
+				Type:     schema.TypeList,
+				Required: true,
+				MinItems: 1,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"arguments": {
+							Type:     schema.TypeMap,
+							Optional: true,
+						},
+						"job_name": {
+							Type:     schema.TypeString,
+							Required: true,
+						},
+						"timeout": {
+							Type:     schema.TypeInt,
+							Optional: true,
+						},
+					},
+				},
+			},
+			"description": {
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+			"enabled": {
+				Type:     schema.TypeBool,
+				Optional: true,
+				Default:  true,
+			},
+			"name": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ForceNew:     true,
+				ValidateFunc: validation.NoZeroValues,
+			},
+			"predicate": {
+				Type:     schema.TypeList,
+				Optional: true,
+				MaxItems: 1,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"conditions": {
+							Type:     schema.TypeList,
+							Required: true,
+							MinItems: 1,
+							Elem: &schema.Resource{
+								Schema: map[string]*schema.Schema{
+									"job_name": {
+										Type:     schema.TypeString,
+										Required: true,
+									},
+									"logical_operator": {
+										Type:     schema.TypeString,
+										Optional: true,
+										Default:  glue.LogicalOperatorEquals,
+										ValidateFunc: validation.StringInSlice([]string{
+											glue.LogicalOperatorEquals,
+										}, false),
+									},
+									"state": {
+										Type:     schema.TypeString,
+										Required: true,
+										ValidateFunc: validation.StringInSlice([]string{
+											glue.JobRunStateFailed,
+											glue.JobRunStateStopped,
+											glue.JobRunStateSucceeded,
+											glue.JobRunStateTimeout,
+										}, false),
+									},
+								},
+							},
+						},
+						"logical": {
+							Type:     schema.TypeString,
+							Optional: true,
+							Default:  glue.LogicalAnd,
+							ValidateFunc: validation.StringInSlice([]string{
+								glue.LogicalAnd,
+								glue.LogicalAny,
+							}, false),
+						},
+					},
+				},
+			},
+			"schedule": {
+				Type:     schema.TypeString,
+				Optional: true,
```

Comment:
are we missing a validation function for cron times here?

Reply:
Hmmm. The AWS API documentation does not provide any guidance on strict validation for `schedule` like some other parameters: https://docs.aws.amazon.com/glue/latest/webapi/API_CreateTrigger.html#API_CreateTrigger_RequestSyntax

It does however point to https://docs.aws.amazon.com/glue/latest/dg/monitor-data-warehouse-schedule.html which we also point to in the Terraform resource documentation.